modules/js-and-css/speculation-rules/helper.php

```diff
@@ -50,7 +50,15 @@ function plsr_get_speculation_rules() {
 		)
 	);
 
-	$prerender_rules = array(
+	$option = get_option( 'plsr_speculation_rules' );
+	if ( ! $option ) {
+		$option = plsr_get_setting_default();
+	}
+
+	$mode      = $option['mode'];
+	$eagerness = $option['eagerness'];
```

Comment:
While unlikely, couldn't it be that perhaps `$option['mode']` isn't set here? If someone put an array into the `plsr_speculation_rules` without setting that key, then either `$option['mode']` or `$option['eagerness']` may not be set. So for extra defensive coding, I think this would be even more robust:

```suggestion
	$option = get_option( 'plsr_speculation_rules' );
	if ( ! is_array( $option ) ) {
		$option = array();
	}
	$option = array_merge( plsr_get_setting_default(), $option );

	$mode      = $option['mode'];
	$eagerness = $option['eagerness'];
```

Reply:
This is extremely unlikely as in the scenario you're describing the value would still always be sanitized at the lowest level (within `update_option()`) to include both values, per the setting registration.

Still, fair enough to do this from a defensive coding perspective. Updated in https://github.com/WordPress/performance/pull/939/commits/d490f9318807937723f92f8e0e005df8266f46f5